pkg/api/types.go

```diff
@@ -448,6 +448,8 @@ const (
 // Host path volumes do not support ownership management or SELinux relabeling.
 type HostPathVolumeSource struct {
 	Path string `json:"path"`
+	// Mount the host path with propagation shared. Docker only.
+	Shared bool `json:"shared"`
```

Comment:
This is not the right type of API for this.  There are multiple propagation modes that we should support:

1.  `shared`
2.  `rshared`
3.  `slave`
4.  `rslave`
5.  `private`

...that means there should be an enum for propagation mode.